### PR TITLE
Hd/ux quality audit some small fixes

### PR DESCRIPTION
**Pull request by RSlysz**
### Purpose of this PR
Various Quality fixes. Including FB https://fogbugz.unity3d.com/f/cases/1341672/ :

List of changes:
- Fix the volume icon location (see FB above)

- better color distinction in shadow cascades for color blindness (using same as debug view mode)
image

- Alignment in LayeredLit material
image

- Missing indentation in Volume Component
image

- ~Disambiguiting the two Film Grain (one URP, one HDRP) by amending the name in "add overrides" and in header~
(See feedback. This is fixed by merging ongoing PR)

- Add infobox when no profile
image

---
### Testing status
Manually tested. 
- Every inspectors looks as expected (see above).
- Removed the HDRP package and Volume Profile and Volume still have their icones displayed.

---
### Comments to reviewers
@Jenniferd as replacement for Alex

### CODE REVIEW

**Review by ellioman — APPROVED** [review]

**Review by jenniferd-unity — APPROVED** [review]
Regarding renaming Film Grain, I think we should also take into account PR #4721 by @alex-vazquez. It looks like both PRs are compatible but maybe it is worth double checking

**Comment by TheoWong-pixel** [comment]
Is it possible to get 'Layer name', 'layer mask', 'UV' in the Layer list to be left aligned so its aligned with its own content?

**Comment by RSlysz** [comment]
Above is @TheoWong-pixel  request:
image

**Review by TheoWong-pixel — APPROVED** [review]

**Comment by TheoWong-pixel** [comment]
@oleks-k @Vic-Cooper FYI on the title chance for Film grain.

File: com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/CameraSettingsUI.Drawers.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ CameraSettingsOverride displayedFields
 
             if ((displayedFields.camera & cullingFields) != 0)
             {
-                PropertyFieldWithoutToggle(CameraSettingsFields.cullingUseOcclusionCulling, serialized.cullingUseOcclusionCulling, EditorGUIUtility.TrTextContent("Use Occlusion Culling"), displayedFields.camera);
+                PropertyFieldWithoutToggle(CameraSettingsFields.cullingUseOcclusionCulling, serialized.cullingUseOcclusionCulling, EditorGUIUtility.TrTextContent("Occlusion Culling"), displayedFields.camera);
                 PropertyFieldWithoutToggle(CameraSettingsFields.cullingCullingMask, serialized.cullingCullingMask, EditorGUIUtility.TrTextContent("Culling Mask"), displayedFields.camera);
                 PropertyFieldWithoutToggle(CameraSettingsFields.cullingInvertFaceCulling, serialized.cullingCullingMask, EditorGUIUtility.TrTextContent("Invert Backface Culling"), displayedFields.camera);
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Editor/Lighting/Reflection/CameraSettingsUI.Drawers.cs:51]: Maybe we can move this `TrTextContent` to static `Styles`?

File: com.unity.render-pipelines.high-definition/Editor/PostProcessing/PaniniProjectionEditor.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+using UnityEditor.Rendering;
+using UnityEngine;
+using UnityEngine.Rendering.HighDefinition;
+
+namespace UnityEditor.Rendering.HighDefinition
+{
+    [VolumeComponentEditor(typeof(PaniniProjection))]
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Editor/PostProcessing/PaniniProjectionEditor.cs:7]: If you merge my PR first, you will need to add the VolumeComponentMenu attribute and specify that this Volume Component is only supported on HDRP, If not, users using URP will be able to add it.

File: com.unity.render-pipelines.universal/Runtime/Overrides/FilmGrain.cs [hunk 3]
```diff
@@ -17,9 +17,14 @@ public enum FilmGrainLookup
         Custom
     }
 
-    [Serializable, VolumeComponentMenu("Post-processing/FilmGrain")]
+    [Serializable, VolumeComponentMenu("Post-processing/Film Grain (URP)")]
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.universal/Runtime/Overrides/FilmGrain.cs:20]: If you merge my PR you will only need the space. remove the (URP)

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Components/FilmGrain.cs [hunk 4]
```diff
@@ -68,10 +68,15 @@ public enum FilmGrainLookup
     /// <summary>
     /// A volume component that holds settings for the Film Grain effect.
     /// </summary>
-    [Serializable, VolumeComponentMenu("Post-processing/Film Grain")]
+    [Serializable, VolumeComponentMenu("Post-processing/Film Grain (HDRP)")]
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Components/FilmGrain.cs:71]: No need of that if you merge my PR

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Components/FilmGrain.cs [hunk 5]
```diff
@@ -68,10 +68,15 @@ public enum FilmGrainLookup
...
     [HDRPHelpURLAttribute("Post-Processing-Film-Grain")]
     public sealed class FilmGrain : VolumeComponent, IPostProcessComponent
     {
+        FilmGrain()
+        {
+            displayName = "Film Grain (HDRP)";
+        }
+
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/Components/FilmGrain.cs:79]: You can get rid of this if you merge my PR

File: com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs [hunk 6]
```diff
@@ -69,19 +69,22 @@ protected override void OnGUIOpen()
...
+            float labelWidth = EditorGUIUtility.labelWidth;
             float height = EditorGUIUtility.singleLineHeight + EditorGUIUtility.standardVerticalSpacing;
 
             EditorGUIUtility.labelWidth = labelWidth;
 
-            Rect headerLineRect = GUILayoutUtility.GetRect(1, height);
-            Rect headerLabelRect = new Rect(headerLineRect.x, headerLineRect.y, EditorGUIUtility.labelWidth - indentOffset + 15f, height);
-            Rect headerUVRect = new Rect(headerLineRect.x + headerLineRect.width - 37f - resetButtonWidth - endOffset, headerLineRect.y, UVWidth + 5, height);
-            Rect headerMaterialDropRect = new Rect(headerLineRect.x + headerLabelRect.width - 20f, headerLineRect.y, headerLineRect.width - headerLabelRect.width - headerUVRect.width, height);
+            Rect headerLineRect = GUILayoutUtility.GetRect(1, 15);
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs:77]: can you add a const variable with the value `15`? in this way we can easily change it.

**Review by alex-vazquez — CHANGES_REQUESTED** [review]
I think is better that we wait to this PR https://github.com/Unity-Technologies/Graphics/pull/4721 to be merged, this will simplify your work

**Review by JulienIgnace-Unity — APPROVED** [review]

File: com.unity.render-pipelines.core/Runtime/Volume/VolumeComponent.cs [hunk 7]
```diff
@@ -94,6 +94,18 @@ public sealed class VolumeComponentDeprecated : Attribute
     [Serializable]
     public class VolumeComponent : ScriptableObject
     {
+        /// <summary>
+        /// Local attribute for VolumeComponent fields only.
+        /// It handles relative indentation of a property for inspector.
+        /// </summary>
+        public sealed class Indent : PropertyAttribute
+        {
+            public readonly int relativeAmount;
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.core/Runtime/Volume/VolumeComponent.cs:103]: missing docs?

File: com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs [hunk 8]
```diff
@@ -65,23 +65,27 @@ protected override void OnGUIOpen()
...
+            const float headerHeight = 15;
 
             EditorGUIUtility.labelWidth = labelWidth;
 
-            Rect headerLineRect = GUILayoutUtility.GetRect(1, height);
-            Rect headerLabelRect = new Rect(headerLineRect.x, headerLineRect.y, EditorGUIUtility.labelWidth - indentOffset + 15f, height);
-            Rect headerUVRect = new Rect(headerLineRect.x + headerLineRect.width - 37f - resetButtonWidth - endOffset, headerLineRect.y, UVWidth + 5, height);
-            Rect headerMaterialDropRect = new Rect(headerLineRect.x + headerLabelRect.width - 20f, headerLineRect.y, headerLineRect.width - headerLabelRect.width - headerUVRect.width, height);
+            Rect headerLineRect = GUILayoutUtility.GetRect(1, headerHeight);
+            Rect headerLabelRect = new Rect(headerLineRect.x, headerLineRect.y, EditorGUIUtility.labelWidth - indentOffset + 15, height);
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs:79]: This 15 is `headerHeight`?

File: com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs [hunk 9]
```diff
@@ -65,23 +65,27 @@ protected override void OnGUIOpen()
...
+            Rect headerUVRect = new Rect(headerLineRect.x + headerLineRect.width - 37 - resetButtonWidth - endOffset, headerLineRect.y, UVWidth + 5, height);
+            Rect headerMaterialDropRect = new Rect(headerLineRect.x + headerLabelRect.width - 15 + 2, headerLineRect.y, headerLineRect.width - headerLabelRect.width - headerUVRect.width, height);
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs:81]: is this `headerHeight` + 2?

File: com.unity.render-pipelines.high-definition/Editor/PostProcessing/FilmGrainEditor.cs [hunk 10]
```diff
@@ -28,7 +28,8 @@ public override void OnInspectorGUI()
 
             if (m_Type.value.intValue == (int)FilmGrainLookup.Custom)
             {
-                PropertyField(m_Texture);
+                using (new IndentLevelScope())
+                    PropertyField(m_Texture);
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.high-definition/Editor/PostProcessing/FilmGrainEditor.cs:32]: You do not need this scope if you are using the attribute, no?

File: com.unity.render-pipelines.universal/Editor/Overrides/FilmGrainEditor.cs [hunk 11]
```diff
@@ -27,7 +27,8 @@ public override void OnInspectorGUI()
 
             if (m_Type.value.intValue == (int)FilmGrainLookup.Custom)
             {
-                PropertyField(m_Texture);
+                using (new IndentLevelScope())
```
Comment (alex-vazquez) [inline com.unity.render-pipelines.universal/Editor/Overrides/FilmGrainEditor.cs:30]: Missing attribute on the component?

**Review by alex-vazquez — CHANGES_REQUESTED** [review]

File: com.unity.render-pipelines.high-definition/Editor/PostProcessing/FilmGrainEditor.cs (hunk 10 above)
Comment (RSlysz) [reply com.unity.render-pipelines.high-definition/Editor/PostProcessing/FilmGrainEditor.cs:32]: We already have the Editor here. I feel the attribute is less instinctive and more hard to read in this case

File: com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs (hunk 9 above)
Comment (RSlysz) [reply com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs:81]: Also indent + horizontal spacing

File: com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs (hunk 8 above)
Comment (RSlysz) [reply com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/LayerListUIBlock.cs:79]: No indent. I'll add another const

File: com.unity.render-pipelines.high-definition/Editor/PostProcessing/FilmGrainEditor.cs (hunk 10 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.high-definition/Editor/PostProcessing/FilmGrainEditor.cs:32]: good, I agree :)

**Review by alex-vazquez — APPROVED** [review]

**Review by erikabar — APPROVED** [review]
Developer already tested changes, approving without additional testing

**Review by sebastienlagarde — APPROVED** [review]

**Pull request merged**